cfep-18.md

```diff
@@ -0,0 +1,60 @@
+
+<table>
+<tr><td> Title </td><td> Uploading binaries from pull requests </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Chris Burr &lt;christopher.burr@cern.ch&gt;</td></tr>
+<tr><td> Created </td><td> June 06, 2020</td></tr>
+<tr><td> Updated </td><td> June 06, 2020</td></tr>
+<tr><td> Discussion </td><td> NA </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+Allow package binaries to be uploaded from builds in pull requests.
+
+## Motivation
+
+Sometimes it would be useful to access binaries built in pull requests. Reasons include:
+
+* Testing GUI applications
+* More easily testing the effects on a pull requests on downstream packages. For example, patches in llvm affecting builds of clang.
+* Debugging non-obvious failures during the tests stage of the build
+
+While it is always possible to build binaries locally it can difficult to set up (e.g. macOS/Windows) or time consuming people only have access to less powerful machines.
+
+## Specification
+
+Upon request, any binaries built in a PR will be uploaded to a file sharing service.
+[Firefox Send](https://github.com/timvisee/ffsend) is proposed in this CFEP however this could change to an alternative service such as [https://transfer.sh/](https://transfer.sh/) or a self hosted equivalent such as [transfer.sh](https://github.com/dutchcoders/transfer.sh) or [0x0](https://github.com/mia-0/0x0).
+
+This change will be implemented by modifying the scripts generated by `conda-smithy` to check for the existence of a file named `CF_UPLOAD_PR_BUILDS` in the recipe directory.
+A similar approach will be used for `staged-recipes`.
+Rerendering will not be required to enable this feature and the `CF_UPLOAD_PR_BUILDS` file will be automatically deleted whenever a feedstock is rerendered.
```

Comment:
Just so I understand, to get the feature in the first place we require at least one rerender. You are saying here that the presence of the file in the feedstock should not affect the CI scripts so that this can be turned on at any time.

Reply:
Yes